studio/src/components/PublishModal/index.tsx

```diff
@@ -1,11 +1,54 @@
 import '@/styles/components/publishModal.scss';
 import { DefaultButton, Icon, Label, Modal, Stack, TextField } from '@fluentui/react';
 import React from 'react';
+import { sendRequest } from '../../api';
+
 interface props {
-    isOpen: boolean
-    hideModal: Function
+    isOpen: boolean,
+    hideModal: Function,
+    representations: any,
+    dslName: string
 }
+
+const uploadProgram = (url: string, secret: string, name: string, representations: any) => {
```

Comment:
This function's arguments list:

'name': 
      'dsl': ,
      'code': ,
      'requirements': '',
      'required_credentials': ,
      'index_urls':
      'version':

Reply:
Preprocessing needs to happen before

Reply:
While this is ideal, let the current pattern exists because I want to move this to the backend.